src/wiki/views/article.py

```diff
@@ -447,82 +448,102 @@ def get_context_data(self, **kwargs):
 
         return super(Move, self).get_context_data(**kwargs)
 
-    # Create a redirect page for src+urlpath[rootlen:] with destination urlpath
-    def create_redirect(self, src, src_slug, root_len, urlpath):
-        dst_path = urlpath.path
-        src_path = src + src_slug + dst_path[root_len:]
-        src_len = len(src_path)
-        pos = src_path.rfind("/", 0, src_len-1)
-        slug = src_path[pos+1:src_len-1]
-        srcurl = models.URLPath.get_by_path(src_path[0:max(pos, 0)])
-
-        try:
-            article = _create_article(
-                self.request, self.article,
-                srcurl, slug,
-                _("Moved: ") + str(urlpath.article),
-                _("Article moved to %s") %
-                ("[wiki:/" + dst_path + "](wiki:/" + dst_path + ")"),
-                "")
-            article.moved_from = urlpath.moved_from
-            article.save()
-            urlpath.moved_from = article
-            urlpath.save()
-            return 1
-        except Exception as e:
-            log.exception("Exception creating redirect article: %s." % str(e))
-            return 0
-
```

Comment:
This whole function was sitting in a strange location and didn't have docstrings, so I just embedded the logic in the view as it isn't shared.